modules/vaos/spec/request/appointments_request_spec.rb

```diff
@@ -10,123 +10,206 @@
+        it 'has an invalid field type exception' do
+          get '/v0/vaos/appointments', params: params.merge(start_date: 'invalid')
+          expect(response).to have_http_status(:bad_request)
+          expect(response.body).to be_a(String)
+          expect(JSON.parse(response.body)['errors'].first['detail'])
+            .to eq('"invalid" is not a valid value for "start_date"')
+        end
+      end
 
-        expect(response).to have_http_status(:success)
-        expect(response.body).to be_a(String)
-        expect(response).to match_response_schema('vaos/va_appointments')
+      context 'with an invalid end_date' do
+        it 'has an invalid field type exception' do
+          get '/v0/vaos/appointments', params: params.merge(end_date: 'invalid')
+          expect(response).to have_http_status(:bad_request)
+          expect(response.body).to be_a(String)
+          expect(JSON.parse(response.body)['errors'].first['detail'])
+            .to eq('"invalid" is not a valid value for "end_date"')
+        end
+      end
+
+      it 'has access and returns va appointments' do
+        VCR.use_cassette('vaos/appointments/get_appointments', match_requests_on: %i[method uri]) do
+          get '/v0/vaos/appointments', params: params
+
+          expect(response).to have_http_status(:success)
+          expect(response.body).to be_a(String)
+          expect(response).to match_response_schema('vaos/va_appointments')
+        end
+      end
+
+      it 'has access and returns cc appointments' do
+        VCR.use_cassette('vaos/appointments/get_cc_appointments', match_requests_on: %i[method uri]) do
+          get '/v0/vaos/appointments', params: params.merge(type: 'cc')
+
+          expect(response).to have_http_status(:success)
+          expect(response.body).to be_a(String)
+          expect(response).to match_response_schema('vaos/cc_appointments')
+        end
       end
     end
 
-    it 'has access and returns cc appointments' do
-      VCR.use_cassette('vaos/appointments/get_cc_appointments', match_requests_on: %i[method uri]) do
-        get '/v0/vaos/appointments', params: params.merge(type: 'cc')
+    describe 'PUT appointments/cancel' do
+      context 'with flipper disabled' do
+        it 'does not have access' do
+          Flipper.disable('va_online_scheduling')
+          put '/v0/vaos/appointments/cancel'
+
+          expect(response).to have_http_status(:forbidden)
+          expect(JSON.parse(response.body)['errors'].first['detail'])
+            .to eq('You do not have access to online scheduling')
+        end
+      end
+
+      context 'when request body validation fails' do
+        it 'returns validation failed' do
+          put '/v0/vaos/appointments/cancel'
+
+          expect(response).to have_http_status(:unprocessable_entity)
+          expect(JSON.parse(response.body)['errors'].size).to eq(2)
+        end
+      end
+
+      context 'when appointment cannot be cancelled' do
```

Comment:
what can't this be cancelled? because the params are wrong?

Reply:
because it is the response the underlying API returns as recorded in that particular cassette.